### PR TITLE
Persistence of models

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- Fix training core model via command line (`rasa train core`)
- Any functions reading model archives is able to handle the case where both or just one model is available.

closes https://github.com/RasaHQ/rasa_nlu/issues/3126

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

File: rasa/test.py [hunk 1]
```diff
@@ -40,14 +40,15 @@ def test_core(model: Text, stories: Text, endpoints: Text = None,
         model_path = get_model(model)
         core_path, nlu_path = get_model_subdirectories(model_path)
 
-        _interpreter = NaturalLanguageInterpreter.create(nlu_path,
-                                                         _endpoints.nlu)
+        if os.path.exists(core_path) and os.path.exists(nlu_path):
```
Comment (tmbo) [inline rasa/test.py]: should we print a message if this `if` isn't entered?

**Comment by tabergma** [comment]
@tmbo Does it make sense to change the scripts in `rasa/nlu` and `rasa/core`? As far as I can see it, the scripts in there can just be called directly via `python rasa/nlu/train.py`etc. - not sure if this is still intended as all functionality can be accessed via the command line interface `rasa` (except the visualize), which calls the scripts in `rasa/nlu` and `rasa/core`.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -18,6 +18,7 @@ Changed
 
 Removed
 -------
+- removed possibility to execute ``python -m rasa_core.train`` etc. (e.g. scripts in ``rasa.core`` and ``rasa.nlu``). Use the CLI for rasa instead, e.g. `rasa train core``.
```
Comment (tmbo) [inline CHANGELOG.rst]: Please also add a note to docs/migration with instructions on how to go from old calls of these scripts to the new cli

File: rasa/core/run.py [hunk 3]
```diff
@@ -195,32 +195,3 @@ def serve_application(core_model=None,
                                action_endpoint=endpoints.action)
 
     return app.agent
-
-
-if __name__ == '__main__':
```
Comment (tmbo) [inline rasa/core/run.py]: I guess we should do the same here as we did it in server.py and raise a runtime error with a link to the migration guide

File: CHANGELOG.rst (hunk 2 above)
Comment (tabergma) [reply CHANGELOG.rst]: can you please point me to the docs/migration where I can add the instructions? thanks.

Comment (tabergma) [reply CHANGELOG.rst]: https://github.com/RasaHQ/rasa_nlu/blob/master/docs/core/migrations.rst ?

**Comment by tabergma** [comment]
@tmbo The documentation points currently to the old scripts, such as 
```
python3 -m rasa.core.run -d models/dialogue -u models/nlu/current --event_broker pika_producer --endpoints endpoints.yml
```
However, even those commands seems to be outdated as there is no option like `--event_broker` anymore. I would like to update the documentation, but I guess I would struggle quite a bit. What do you think? Will this be tackled anyway with the updated documentation?

**Comment by tmbo** [comment]
Yes, we need to rewrite these commands. The question is how much that will interfere with the documentation work @amn41 is currently doing. I'd want to avoid doing to many docs changes while he is working on that. @amn41 would it be an issue to change the commands now or should we rather create an issue and tackle it after your work is done?

**Comment by amn41** [comment]
please re-write now so the docs are up to date, I'll manage any conflicts on my end

File: docs/migrations.rst [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+0.14.x to 0.15.0
+
+General
+~~~~~~~
+
+- The scripts in ``rasa.core`` and ``rasa.nlu`` can no longer be executed. To train, test, run, ... a rasa nlu or core
+  model, you should now use the command line interface ``rasa``. The functionality is the same as before. If you run
+  one of the old scripts in ``rasa.core`` or ``rasa.nlu`` an error is thrown that also points you to the command you
+  should use instead.
```
Comment (tmbo) [inline docs/migrations.rst]: please also add a mapping from old to new script here for reference.

**Pull request merged**